### DISCUSSION TITLE
Pomiary PMS7003 vs SDS011

### DISCUSSION THREAD

**Discussion by skwaruch** [hackerspace-silesia/Smogomierz#144] [General]
Do przemyślenia :) (spora różnica)
SDS011:
![image](https://user-images.githubusercontent.com/12412153/67139332-5e4db580-f24f-11e9-9a61-86c2852439c9.png)
PMS7003:
![image](https://user-images.githubusercontent.com/12412153/67139356-acfb4f80-f24f-11e9-9ee6-88712d02ae4b.png)

**Comment by bfaliszek** [comment]
PMS7003:
"±10% @ 100 ~ 500 μg/m³
±10μ g/m³ @ 0 ~ 100 μg/m³"
[PMS7003_manual.pdf](https://github.com/hackerspace-silesia/Smogomierz/files/3746439/PMS7003_manual.pdf)

SDS011:
"Maximum of ± 15% and ±10μg/m3"
[SDS011_laser_PM2.5_sensor_specification-V1.3.pdf](https://github.com/hackerspace-silesia/Smogomierz/files/3746440/SDS011_laser_PM2.5_sensor_specification-V1.3.pdf)

**Comment by hali71** [comment]
Mam pytanie o kalibracje czujnika. Jakie jest źródło obliczania współczynnika calib:
```
      if (int(BMESensor.temperature) < 5 or int(BMESensor.humidity) > 60) {
        calib1 = float((200 - (BMESensor.humidity)) / 150);
        calib2 = calib1 / 2;
        calib = calib2;
      } else {
        calib = calib1;
```
Szukałem jakiegos potwierdzenia tej kalibracji w necie ale niestety nie znalazłem. A od jakiegoś czasu zastanawiam się jak korygować pomiary np podczas duzej wilgotności lub mgle. Uzywam czujnika PMS7003.

**Comment by bfaliszek** [comment]
calib obliczany jest na podstawie pomiarów wilgotności powietrza. Później pomiary z PMS są mnożone razy ten współczynnik. Wypracowałem go na podstawie pomiarów z kilku różnych urządzeń pomiarowych, w tym profesjonalnych pożyczonych od alarmu smogowego.

**Comment by black-red-dot** [comment]
Czy wspołczynnik kalibracji obliczany jest w ten sposób:
`float((200 - (BMESensor.humidity)) / 150);` calib1
czy jeszcze dodatkowo dzielony jest przez 2:
`float((200 - (BMESensor.humidity)) / 150)/2;` calib2

z kodu, od miejsca:
`// Automatic calibration - END`
`if (!strcmp(THP_MODEL, "BME280")) {
    calib = calib1;`
wynika, że dodatkowe dzielenie jest pomijane

**Comment by bfaliszek** [comment]
Te współczynniki obliczane są zawsze, ale w zależności od ustawień, są używane albo nie. Choć nie do końca wiem, o który fragment kodu Ci chodzi. Zamiast tak cytować kawałek linijki, możesz wrzucić linka do niej w konkretnym pliku w repozytorium. Wtedy będę bardziej wiedział o które miejsce chodzi.

**Comment by black-red-dot** [comment]
Chodzi mi to, że najpierw współczynnik calib przyjmuje wartość calib2 [https://github.com/hackerspace-silesia/Smogomierz/blob/master/Smogomierz.ino#L1890-L1892](url) ale za chwilę zawsze zmieniany jest na calib1 [https://github.com/hackerspace-silesia/Smogomierz/blob/master/Smogomierz.ino#L1926-L1927](url)
